common/src/main/java/com/hortonworks/streamline/common/SchemaValueConverter.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2017 Hortonworks.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *   http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package com.hortonworks.streamline.common;
+
+import com.hortonworks.registries.common.Schema;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toMap;
+
+public final class SchemaValueConverter {
+    private SchemaValueConverter() {
+    }
+
+    public static Map<String, Object> convertMap(Schema schema, Map<String, Object> value) {
+        Map<String, Schema.Type> fieldToType = schema.getFields().stream()
+                .collect(toMap(Schema.Field::getName, Schema.Field::getType));
+
+        boolean fieldNotFound = (schema.getFields().size() != value.keySet().size()) ||
+                value.keySet().stream().anyMatch(f -> !(fieldToType.containsKey(f)));
+        if (fieldNotFound) {
+            throw new IllegalArgumentException("Field list doesn't match to output schema.");
+        }
+
+        return value.entrySet().stream()
+                .collect(toMap(Map.Entry::getKey, e -> convert(fieldToType.get(e.getKey()), e.getValue())));
+    }
+
+    public static Object convert(Schema.Type type, Object value) {
+        if (value == null) {
+            return null;
+        }
+
+        switch (type) {
+            case BOOLEAN:
+                if (value instanceof Boolean) {
+                    return value;
+                } else if (value instanceof String) {
+                    return Boolean.valueOf((String) value);
+                }
+                break;
+
+            case BYTE:
+            case SHORT:
+            case INTEGER:
+            case LONG:
+            case FLOAT:
+            case DOUBLE:
+                return convertNumber(type, value);
+
+            case STRING:
+                if (value instanceof String) {
+                    return value;
+                }
+                break;
+
+            case BINARY:
+                if (value instanceof byte[]) {
+                    return value;
+                }
+                break;
+
+            case NESTED:
+                if (value instanceof Map) {
+                    return value;
+                }
+                break;
+
+            case ARRAY:
+                if (value instanceof List) {
+                    return value;
+                }
+                break;
+
+        }
+
+        throw new IllegalArgumentException("Cannot convert value " + value + " with Java type class " +
```

Comment:
Can this be moved under the default case?

Reply:
There're some `cases` which uses the line, so can't move to the default case. Default case would reach here too.